src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+
+import { Button } from '../Button';
+import { Input } from '../Input';
+import { Tag } from '../Tag';
+
+import {
+    DivStyled,
+    DivStyledContent,
+    SectionStyled,
+} from './Todo.styles';
+
+import { TodoProps, TodoState } from './types';
+
+const Todo: React.FC<TodoProps> = ({ todos = [], pattern, fullWidth = false }) => {
+    const [inputValue, setInputValue] = useState<string>('');
+    const [lists, setLists] = useState<TodoState[]>([]);
+
+    useEffect(() => {
+        if (Array.isArray(todos) && todos?.length > 0) {
+            setLists([...todos]);
+        }
+    }, [todos]);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setInputValue(e.target.value);
+    };
+
+    const onCancelClick = (id: number) => {
+        setLists(lists.filter((todo: TodoState) => todo.id !== id));
+    };
+
+    const addTodo = () => {
+        const todo = {
+            id: Date.now(),
+            text: inputValue,
+        };
+        setLists((prevTodo) => [...prevTodo, todo]);
+        setInputValue('');
+    };
+
+    return (
+        <SectionStyled data-testid="todo-comp">
+            <DivStyled>
+                <Input
+                    label="Enter IP"
```

Comment:
make it modular

Reply:
is there a need to accept button text ?

Reply:
not really but could be handy in future. Up to you my bro

Reply:
okay